vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.jsp

```diff
@@ -48,38 +47,60 @@ function onDoubleClick(e){
         // remove highlighting if present
         $(this).removeClass("hova");
     } else {
-        $("#addImgAlert").show();
+    	
+    	// storing image ID selected by the user to replace, onDoubleClick
+    	var imgID = $(e.target).attr('id'); 
+    	$("#uploadImage").data('value', imgID) // sets image ID value
+    	$("#addImgAlert").show();
+       	
     }
 }
     
 function uploadImage() {
-    var file = document.getElementById('file').files[0];
+    
+	var imgID = $("#uploadImage").data('value') // gets image ID
+    var file = $('#file')[0].files[0]
     var reader  = new FileReader();
     var formData = new FormData();
     formData.append('file', file);
     formData.append('contentOrder', contentCount);
-    if ($(".open")[0]){
-        var imageBlockId = $('.open img').attr('id')
+    var imageblock = "";
+     
+    // Ashmi changes for Story VSPC-64
+    
+	// checks if image ID is present to replace
+    if (imgID != '') {
+    
+    	var imageBlockId = imgID;
         formData.append('imageBlockId',imageBlockId);
         var url = "<c:url value="/staff/module/${module.id}/slide/${slide.id}/image/" />" + imageBlockId + "?${_csrf.parameterName}=${_csrf.token}";
-        reader.onload = function () {
-            imageblock = createImageBlock(reader);
-            $("#" + imageBlockId).replaceWith(imageblock);
+        reader.onload = function (theFile) {
+        	
+	        var image = new Image();
+	        image.src = theFile.target.result;
+	        image.onload = function () {
+	            imageblock = createImageBlock(reader, this.width);
+	            $("#" + imageBlockId).attr("id", imageBlockId);
+	            $("#" + imageBlockId).replaceWith(imageblock);
```

Comment:
why do you set the id in line 83 if you then replace that element in line 84?

Reply:
(Uncaught TypeError: imageblock.find is not a function)
I am not getting any error on console of this type.

Also, I am able to change a changed Image on double-click.

I will fix Line 83.